rfcs/0000_guidance_on_using_llms.md

```diff
@@ -0,0 +1,59 @@
+- We want to get a clear understanding of your solution, and we have found that LLMs give poor, if approximately correct, RFC proposals. This is particularly the case because the work of the TBP falls very much in the "out of training distribution" domain.
+- When we review RFCs written by LLMs, it often ends up taking more time both for us, and for you, so it is much better if you write them yourself.
+
+Note that it is acceptable to discuss ideas with an LLM when formulating your own thoughts. For example, there may be a concept in neuroscience or computer vision that you would like to understand better, and LLM conversations can be a useful starting point. However, the RFC should be written in your own words. Furthermore, we recommend that you always be wary of hallucinations in LLM outputs; due to their frequent occurrence, LLMs are better used as an initial - but not final - provider of information on a topic.
+
+## Contributing Code to `tbp.monty`
+
+In principle, you may use LLMs, such as code-assistants, when writing code that you contribute to the Thousand Brains Project. However, we ask that you do so in a limited manner, being mindful of the below issues:
+
+### Quality Concerns
+- As with RFCs, we have found that our code-base is out-of-distribution, and the quality of code written by LLMs, while superficially correct, often is not.
+- Similarly, Monty is fundamentally about developing sensorimotor AI that can succeed in embodied settings. These are precisely the settings that LLMs struggle to perform in, and so once again we have found they frequently provide incorrect solutions.
+- LLM code is often verbose, or results in excessive, unnecessary changes to the code.
+
+Due to the above reasons, LLM-generated code can take a great deal of time to review and debug. This can be avoided when PRs are written with intent by a person.
+
+### Legal Concerns
+- There are non-trivial legal concerns around contamination of code when using LLMs. See for example, [this recent study](https://arxiv.org/html/2408.02487v1), which demonstrates that LLM-generated code can violate copyright restrictions in a significant number of cases.
+- A cornerstone of the Thousand Brains Project is its open source nature, and this motivates our use of an MIT licence when we distribute our code. However, the inadvertent integration of copyright-protected code into `thousandbrainsproject/tbp.monty` could jeopardize the ability to make the code open-source, disrupting any distributions of the platform.
+
+### Take-Aways
+
+The high-level guidance based on the above is:
+- Using an LLM to auto-complete variable names and other simple speed-ups can be appropriate.
+- Multi-line sections of algorithmic code written by LLMs should be thoroughly checked for logical correctness and potential copyright violations before opening a PR into `thousandbrainsproject/tbp.monty`.
```

Comment:
thought: It would be nice if we could provide some concrete guidance on how to check for legal compliance. @codeallthethingz did you have any more luck looking into this?